captcha-plugin-draft.txt

```diff
@@ -333,6 +347,9 @@ Appendix D: Security Considerations [SECURITY]
     particular IP) and save the resulting tokens for later.
 
     [TODO: We don't have a great answer for this. Halp!]
+    [Alex: Might be possible to enforce the plugin to keep only 10,000 nonces per
+    pinned challenge key and then check that the plugin is unmodified by checking
+    that the Tor browser is unmodified in some way (or having a signed plugin?).]
```

Comment:
See above :) We're absolutely relying on assumptions about the integrity of Tor Browser.